packages/arb-avm-cpp/avm/src/machinestate/machineoperation.cpp

```diff
@@ -900,13 +900,16 @@ void send(MachineState& m) {
 }
 
 BlockReason inboxOp(MachineState& m) {
-    MachineMessage next_message;
-    if (!m.context.inboxEmpty()) {
-        next_message = m.context.popInbox();
-    } else {
+    if (m.context.inboxEmpty()) {
         return InboxBlocked();
     }
 
+    auto next_message = m.context.popInbox();
+
+    if (next_message.message.timestamp > m.last_inbox_timestamp) {
+        m.l1_block_number = next_message.message.block_number;
+        m.last_inbox_timestamp = next_message.message.timestamp;
+    }
```

Comment:
Might want a separate check for the block number being greater in case one goes up and the other down

Reply:
:+1: